lib/rules/name-property-required.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * @fileoverview Require a name property in Vue components
+ * @author LukeeeeBennett
+ */
+'use strict'
+
+const utils = require('../utils')
+
+function isNameProperty (node) {
+  return node.type === 'Property' && node.key.name === 'name'
+}
+
+function hasTruthyLiteralValue (node) {
+  return node.value.type === 'Literal' && node.value.value
```

Comment:
I think that another rule is necessary to verify whether it is Literal.
What do you think?

Reply:
Sounds like a good idea yes :) I will open a different PR for it and link here